snapshot/snapshot.cc

```diff
@@ -0,0 +1,73 @@
+#include "snapshot.h"
+#include <fstream>
+#define MTIMECMP_BASE	0x4000
+#define MTIME_BASE	0xbff8
```

Comment:
Extract MTIMECMP_BASE and MTIME_BASE into header file and then use that would be better, those two value already defined in clint.cc.

Reply:
OK. We'll extract these into a header file.